libcontainer/intelrdt/intelrdt.go

```diff
@@ -166,22 +205,38 @@ type Manager interface {
 
 // This implements interface Manager
 type intelRdtManager struct {
-	mu     sync.Mutex
-	config *configs.Config
-	id     string
-	path   string
+	mu              sync.Mutex
+	config          *configs.Config
+	id              string
+	path            string
+	groupType       string
+	switchGroupType func(m *intelRdtManager, getIntelRdtPath func(string, string) (string, error)) error
 }
 
 func NewManager(config *configs.Config, id string, path string) Manager {
+	var groupType string
+	if config.IntelRdt.L3CacheSchema != "" || config.IntelRdt.MemBwSchema != "" {
+		groupType = ControlGroupType
+	} else {
+		groupType = MonitoringGroupType
+	}
 	return &intelRdtManager{
-		config: config,
-		id:     id,
-		path:   path,
+		config:          config,
+		id:              id,
+		path:            path,
+		groupType:       groupType,
+		switchGroupType: switchGroup,
```

Comment:
If based on the change in https://github.com/opencontainers/runc/pull/2523/files#r514121535, we don't need the member function `switchGroupType()` initialization here.